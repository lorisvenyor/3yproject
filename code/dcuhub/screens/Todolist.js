import React from "react";
import { StyleSheet, Text, View, TouchableOpacity, FlatList, Modal, Alert, Dimensions} from "react-native";
import { AntDesign } from "@expo/vector-icons";
import colors from "../shared/Colors";
import TodoList from "../components/TodoList";
import AddListModal from "../components/AddListModal";
import Fire from "../config/Fire";
import {SafeAreaView} from 'react-native-safe-area-context'
import { StatusBar } from 'expo-status-bar';
import { ScrollView } from "react-native-gesture-handler";


export default class Todolists extends React.Component {
    state = {
        addTodoVisible: false, // setting task automatically to false
        lists: [],
        user: {},
        
    };

    componentDidMount() {
        firebase = new Fire((error, user) => {
            if (error) {
                return alert("Error");
            }

            firebase.getLists(lists => {
                this.setState({ lists, user }, () => {
                    
                });
            });

            this.setState({ user });
        });
    }

    componentWillUnmount() {
        firebase.detach();
    }

    toggleAddTodoModal() {
        this.setState({ addTodoVisible: !this.state.addTodoVisible });
    }

    renderList = list => {
        return <TodoList list={list} updateList={this.updateList} deleteList={this.deleteList}  />
      }

    addList = list => { // ading list name and todo's etc to firestore
        firebase.addList({
            name: list.name,
            color: list.color,
            todos: []
        })
    }

    deleteList = list => { // deleting list in firestore
       
        firebase.deleteList(list); 
      }

    updateList = list => { // updating list in firestore
        firebase.updateList(list);
    }

    render() {
     
        return (
            <SafeAreaView style={styles.container}>
            <StatusBar style="auto"/>
            <ScrollView>

            <View style={{
                alignItems: "center",
                justifyContent: "center"
            }}>            

                <Modal
                    animationType="fade"
                    visible={this.state.addTodoVisible}
                    onRequestClose={() => this.toggleAddTodoModal()}>
                    <AddListModal closeModal={() => this.toggleAddTodoModal()} addList={this.addList} />
                </Modal>

                <View style={{ marginVertical: 50 }}>
                    <TouchableOpacity style={styles.addList} onPress={() => this.toggleAddTodoModal()}>
                        <AntDesign 
                        name="plus"
                         size={25} 
                         color={colors.orange} />
                    </TouchableOpacity>
                    <Text style={styles.add}>Add new list</Text>
                </View>

                <View style={{height: 400}}>
                    <FlatList
                        data={this.state.lists}
                        keyExtractor={item => item.id.toString()}
                        horizontal={true}
                        showsHorizontalScrollIndicator={false}
                        renderItem={({ item }) => this.renderList(item)}
                        keyboardShouldPersistTaps="always"
                    />
                </View>
            
            </View>
            </ScrollView>
            </SafeAreaView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "white",
    },
    divider: {
        backgroundColor: colors.orange,
        height: 1,
        flex: 1,
        alignSelf: "center"
    },
   
    addList: {
        borderWidth: 8,
        borderColor: colors.orange,
        borderRadius: 20,
        padding: 20,
        alignItems: "center",
        justifyContent: "center"
    },
    add: {
        color: colors.orange,
        fontWeight: '500',
        fontSize: 18,
        marginTop: 15
    },
    button_orange: {
        paddingHorizontal: 20,
        alignItems: "center",
        justifyContent: "center",
        marginTop: 30,
        backgroundColor: "orange",
        paddingVertical: 8,
        borderRadius: 20
    },
    button_white: {
        alignItems: "center",
        justifyContent: "center",
        borderWidth: 2,
        borderColor: "orange",
        paddingVertical: 5,
        borderRadius: 20,
        paddingHorizontal: 30,
        marginHorizontal: 50,
    }
})