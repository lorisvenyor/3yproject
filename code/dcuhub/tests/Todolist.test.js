import React from 'react'
import {create, act} from 'react-test-renderer'
import Todolist from '../screens/Todolist'

window.addEventListener = jest.fn()
window.attachEvent = jest.fn()

const screen = create(<Todolist/>)

test('Todolist Screen Snapshot', () => {
    expect(screen).toMatchSnapshot()
})

