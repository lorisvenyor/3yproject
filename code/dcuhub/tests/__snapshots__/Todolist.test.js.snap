// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todolist Screen Snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Modal
          animationType="fade"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          visible={false}
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 32,
                  "top": 64,
                }
              }
            >
              <Text />
            </View>
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "marginHorizontal": 32,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#2D3436",
                    "fontSize": 28,
                    "fontWeight": "800",
                    "marginBottom": 16,
                  }
                }
              >
                Create To-Do List
              </Text>
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="List Name"
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#24A6D9",
                    "borderRadius": 6,
                    "borderWidth": 0.5,
                    "fontSize": 18,
                    "height": 50,
                    "marginTop": 8,
                    "paddingHorizontal": 16,
                  }
                }
                underlineColorAndroid="transparent"
              />
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginHorizontal": 50,
                    "marginTop": 15,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#24A6D9",
                      "borderRadius": 5,
                      "height": 30,
                      "opacity": 1,
                      "width": 30,
                    }
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#FFBD4A",
                      "borderRadius": 5,
                      "height": 30,
                      "opacity": 1,
                      "width": 30,
                    }
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#f44a6e",
                      "borderRadius": 5,
                      "height": 30,
                      "opacity": 1,
                      "width": 30,
                    }
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#9224ef",
                      "borderRadius": 5,
                      "height": 30,
                      "opacity": 1,
                      "width": 30,
                    }
                  }
                />
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFBD4A",
                    "borderRadius": 6,
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 24,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontWeight": "600",
                    }
                  }
                >
                  Create
                </Text>
              </View>
            </View>
          </View>
        </Modal>
        <View
          style={
            Object {
              "marginVertical": 50,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#efa024",
                "borderRadius": 20,
                "borderWidth": 8,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 20,
              }
            }
          >
            <Text />
          </View>
          <Text
            style={
              Object {
                "color": "#efa024",
                "fontSize": 18,
                "fontWeight": "500",
                "marginTop": 15,
              }
            }
          >
            Add new list
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 400,
            }
          }
        >
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            keyboardShouldPersistTaps="always"
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
